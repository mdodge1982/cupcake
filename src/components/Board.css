.Board {
	position: relative;
	overflow: hidden;
	border: 3px solid black;
	background:
	radial-gradient(white 10%, transparent 11%),
	radial-gradient(white 10%, transparent 11%),
	#efefef;
	background-position: 0 0, 70px 70px;
	-webkit-background-size:140px 140px;
	-moz-background-size:140px 140px;
	background-size:140px 140px;
}
@media screen and ( max-height: 500px ){
	.holder {
		height: 120%;
	}
	.Board {
		height: 100%;
		border: none;
	}
}
.padding {
	padding: 10px;
}
.intro {
	margin: 20px auto;
	text-align: left;
	width: 100%;
	max-width: 700px;
	padding: 0 10px 0 70px;
	background: url('images/bag_front.png') 10px 10px no-repeat;
}
.message {
	display: inline-block;
	text-align: left;
	height: 110px;
	padding: 40px 20px 0 60px;
	margin: 0 0 10px;
	background: url('images/bag_front.png') left 10px no-repeat;
}
.lose {
	background-image: url('images/bag_frown.png');
}
