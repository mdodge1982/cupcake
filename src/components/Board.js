import React, {Component} from 'react';
import ControlsContainer from '../containers/ControlsContainer';
import FrostingContainer from '../containers/FrostingContainer';
import CupcakesContainer from '../containers/CupcakesContainer';
import BagContainer from '../containers/BagContainer';
import Instructions from '../components/Instructions';
import Receptacle from '../components/Receptacle';
import './Board.css';

class Board extends Component {
	constructor() {
		super();
		this.state = {
			int: null
		}
	}
	render() {
		const {width,height,gameStarted,gameOver,trash,box} = this.props;
		let message = 'Not bad.';
		let messageClass = 'message';
		const lose = trash.count>=box.count;
		if(lose){
			message = 'Not great.';
			messageClass += ' lose';
		}else if(box.count-trash.count>3){
			message = 'Nicely done!'
		}
		return (
			<div style={{width:width+'px'}}>
				<h1>Cupcake Invaders</h1>
				<div className="Board" style={{height:height+'px'}}>
					{!gameStarted &&
						<div className="intro">
							<h2>Cupcake Invaders</h2>
							<h3>Frost as many cupcakes as you can!</h3>
							<Instructions />
							<p>You can also use the <strong>&lt; o &gt;</strong> buttons at the bottom of the board to move and shoot.</p>
							<p><button className="play" onClick={() => this.handleClick()}>Play Now</button></p>
						</div>
					}
					{gameOver &&
						<div className="padding">
							<h4 className={messageClass}>{message}</h4>
							<Receptacle {...trash} inline={true} />
							&nbsp; &nbsp; &nbsp;
							<Receptacle {...box} inline={true} />
							<p>&nbsp;</p>
							<p><button className="play" onClick={() => this.handleClick()}>Play Again</button></p>
						</div>
					}
					{gameStarted&&!gameOver &&
						<div>
							<CupcakesContainer />
							<Receptacle {...trash} />
							<Receptacle {...box} />
							<FrostingContainer />
							<BagContainer />
						</div>
					}
				</div>
				{gameStarted &&
				<ControlsContainer />
				}
			</div>
		);
	}
	componentWillUnmount() {
		clearInterval(this.state.int);
	}
	handleClick() {
		this.props.startGame();
		clearInterval(this.state.int);
		this.setState({int:setInterval(() => {
			if(!this.props.gameOver){
				this.props.moveObjects();
			}else{
				clearInterval(this.state.int);
			}
		},25)});
	}
}

export default Board;
