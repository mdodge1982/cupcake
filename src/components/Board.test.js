import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import renderer from 'react-test-renderer';
import Board from './Board';

const board = {
	width: 667,
	height: 430,
	gameStarted: false,
	gameOver: false,
	trash: {count: 0},
	box: {count: 0}
};
const cupcakes = {
	byId: {},
	visibleIds: []
};
const frosting = {
	byId: {},
	visibleIds: []
};
const reducer = (state = {board,cupcakes,frosting}) => state;
const store = createStore(reducer);

const testSnapshot = () => {
	const component = renderer.create(
		<Provider store={store}>
			<Board {...board} />
		</Provider>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
};

test('Snapshot test game not started', () => {
	testSnapshot();
});

test('Snapshot test game started', () => {
	board.width = 685;
	board.height = 398;
	board.gameStarted = true;
	testSnapshot();
});

test('Snapshot test game OVER', () => {
	board.gameOver = true;
	board.trash.count = 5;
	board.box.count = 14;
	testSnapshot();
});

jest.useFakeTimers();

test('onClick test', () => {
	const mockStart = jest.fn();
	const mockMove = jest.fn();
	board.gameOver = false;
	board.gameStarted = false;
	const component = ReactTestUtils.renderIntoDocument(
		<Board store={store} startGame={mockStart} moveObjects={mockMove} {...board} />
	);
	const button = ReactTestUtils.findRenderedDOMComponentWithClass(component,'play');
	ReactTestUtils.Simulate.click(button);
	jest.runTimersToTime(26);
	clearInterval(component.state.int);
	expect(mockStart.mock.calls.length).toBe(1);
	expect(mockMove.mock.calls.length).toBe(1);
});
