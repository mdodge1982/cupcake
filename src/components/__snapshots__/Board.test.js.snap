// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test game OVER 1`] = `
<div
  style={
    Object {
      "width": "685px",
    }
  }
>
  <h1>
    Cupcake Invaders
  </h1>
  <div
    className="Board"
    style={
      Object {
        "height": "398px",
      }
    }
  >
    <div
      className="padding"
    >
      <h4
        className="message"
      >
        Nicely done!
      </h4>
      <div
        className="Receptacle undefined"
        style={
          Object {
            "display": "inline-block",
            "height": "undefinedpx",
            "position": "static",
            "width": "undefinedpx",
          }
        }
      >
        5
      </div>
           
      <div
        className="Receptacle undefined"
        style={
          Object {
            "display": "inline-block",
            "height": "undefinedpx",
            "position": "static",
            "width": "undefinedpx",
          }
        }
      >
        14
      </div>
      <p>
         
      </p>
      <p>
        <button
          className="play"
          onClick={[Function]}
        >
          Play Again
        </button>
      </p>
    </div>
  </div>
  <div
    className="Controls"
  >
    <button
      className="Arrow left"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <button
      className="shoot"
      onClick={[Function]}
    />
    <button
      className="Arrow right"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <p
      className="Instructions small"
    >
      <strong>
        HINT:
      </strong>
       Use the 
      <strong>
        left
      </strong>
       and 
      <strong>
        right
      </strong>
       arrow keys to move your pastry bag, and the 
      <strong>
        spacebar
      </strong>
       (or click anywhere on the board) to shoot frosting.
    </p>
  </div>
</div>
`;

exports[`Snapshot test game not started 1`] = `
<div
  style={
    Object {
      "width": "667px",
    }
  }
>
  <h1>
    Cupcake Invaders
  </h1>
  <div
    className="Board"
    style={
      Object {
        "height": "430px",
      }
    }
  >
    <div
      className="intro"
    >
      <h2>
        Cupcake Invaders
      </h2>
      <h3>
        Frost as many cupcakes as you can!
      </h3>
      <p
        className="Instructions"
      >
        
         Use the 
        <strong>
          left
        </strong>
         and 
        <strong>
          right
        </strong>
         arrow keys to move your pastry bag, and the 
        <strong>
          spacebar
        </strong>
         (or click anywhere on the board) to shoot frosting.
      </p>
      <p>
        You can also use the 
        <strong>
          &lt; o &gt;
        </strong>
         buttons at the bottom of the board to move and shoot.
      </p>
      <p>
        <button
          className="play"
          onClick={[Function]}
        >
          Play Now
        </button>
      </p>
    </div>
  </div>
</div>
`;

exports[`Snapshot test game started 1`] = `
<div
  style={
    Object {
      "width": "685px",
    }
  }
>
  <h1>
    Cupcake Invaders
  </h1>
  <div
    className="Board"
    style={
      Object {
        "height": "398px",
      }
    }
  >
    <div>
      <ul
        className="Cupcakes"
      />
      <div
        className="Receptacle undefined"
        style={
          Object {
            "bottom": "undefinedpx",
            "height": "undefinedpx",
            "left": "undefinedpx",
            "width": "undefinedpx",
          }
        }
      >
        0
      </div>
      <div
        className="Receptacle undefined"
        style={
          Object {
            "bottom": "undefinedpx",
            "height": "undefinedpx",
            "left": "undefinedpx",
            "width": "undefinedpx",
          }
        }
      >
        0
      </div>
      <ul
        className="Frosting"
        onClick={[Function]}
      />
      <div
        className="Bag facing-undefined"
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "height": "undefinedpx",
            "left": "undefinedpx",
            "transform": "rotate(NaNrad)",
            "width": "undefinedpx",
          }
        }
      />
    </div>
  </div>
  <div
    className="Controls"
  >
    <button
      className="Arrow left"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <button
      className="shoot"
      onClick={[Function]}
    />
    <button
      className="Arrow right"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    />
    <p
      className="Instructions small"
    >
      <strong>
        HINT:
      </strong>
       Use the 
      <strong>
        left
      </strong>
       and 
      <strong>
        right
      </strong>
       arrow keys to move your pastry bag, and the 
      <strong>
        spacebar
      </strong>
       (or click anywhere on the board) to shoot frosting.
    </p>
  </div>
</div>
`;
